src/Serilog/Core/LevelOverrideMap.cs

```diff
@@ -68,7 +68,7 @@ public LevelOverride(string context, LoggingLevelSwitch levelSwitch)
     {
         foreach (var levelOverride in _overrides)
         {
-            if (context.StartsWith(levelOverride.Context) &&
+            if (context.StartsWith(levelOverride.Context, StringComparison.OrdinalIgnoreCase) &&
                 (context.Length == levelOverride.Context.Length || context[levelOverride.Context.Length] == '.'))
```

Comment:
Surely the second part of this condition is more selective and cheaper to compute ?
i.e.
            var loLen = levelOverride.Context.Length;
            if ((context.Length == loLen || (context.Length > loLen && context[loLen] == '.'))
                && context.StartsWith(levelOverride.Context, StringComparison.OrdinalIgnoreCase))
                
(my C# is too rusty to convert that to an `is` expression here!)

Reply:
Oh, yes, good call @bartelink. The clause would need a slight modification:

```csharp
            if (context.Length >= levelOverride.Context.Length &&
                (context.Length == levelOverride.Context.Length || context[levelOverride.Context.Length] == '.') &&
                context.StartsWith(levelOverride.Context, StringComparison.OrdinalIgnoreCase))
```

@tillig I wonder if this would be enough to claw back a few % of the performance regression related to the case-sensitivity check? I'll dig in and take a look when I have a chance, if you don't beat me to it :-)

Reply:
I'm out of town for a week or so, so it'll be a bit before I'm back here. Happy to make the change, just will be a bit.

Reply:
I'm back and got this. Sorry for the delay.

Reply:
Hmmm. It appears swapping those has started causing tests to fail. I'll have to look into it.

Reply:
OK, I'm dumb, I missed the thing @nblumhardt mentioned.

I split the `==` and `>` so in the full match scenario it should be just slightly faster since it won't have to check for equality again. I'm sure it's on the order of microseconds, but 🤷 .